arch/riscv/core/xuantie/Kconfig

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2025 MASSDRIVER EI (massdriver.space)
+# SPDX-License-Identifier: Apache-2.0
+
+config XUANTIE_E907
+	bool "XuanTie E907 Core"
```

Comment:
this is a soc not an arch, this does not go here

Reply:
@nordicjm Could you update on this please? Would like to move forward.

Reply:
@tejlmand Please take a look, this is blocking all further PRs.

Reply:
@fkokosinski Could you give us your opinion on this please? I'd like to move forward, this is blocking current and future PRs, Thank you.

Reply:
> I feel XuanTie is somewhere between arch and SoC

That is similar to what I initially thought about arch, so would you recommend what JarmouniA is suggesting (core root folder), keeping it in riscv/core (i think that makes sense because then arch/*/core can be reserved for when there are many uArchs like for riscv without stepping on the toes of 'actual' arch things), or something like soc/common/core ?

Reply:
> #97130

Thanks for creating this.